app/services/immunity_details_creation_service.rb

```diff
@@ -22,4 +23,27 @@ def call
   def application_end_date
     @planning_application.find_proposal_detail("When were the works completed?").first.response_values.first
   end
+
+  def create_evidence_groups
+    Document::EVIDENCE_TAGS.each do |tag|
+      next if @planning_application.documents.with_tag(tag).empty?
+
+      @planning_application.documents.with_tag(tag).each do |doc|
+        @planning_application.immunity_detail.add_document(doc)
+      end
+    end
+
+    @planning_application.immunity_detail.evidence_groups.each do |eg|
+      Document::EVIDENCE_QUESTIONS[eg.tag.to_sym].each do |question|
+        if question.include? "show"
+          eg.applicant_comment = @planning_application.find_proposal_detail(question).first.response_values.first
+        elsif (question.include?("start") || question.include?("issued"))
+          eg.start_date = @planning_application.find_proposal_detail(question).first.response_values.first
+        elsif question.include? "run"
+          eg.end_date = @planning_application.find_proposal_detail(question).first.response_values.first
+        end
+        eg.save
```

Comment:
Do we wanna use `save!` here since we want to raise an error instead of silently failing